src/field_theory/ratfunc.lean

```diff
@@ -1243,6 +1259,89 @@ end
 
 end eval
 
+section int_degree
+
+open polynomial
+
+omit hring
+variables [hfield : field K]
+include hfield
+
+/-- `int_degree x` is the degree of the rational function `x`, defined as the difference between
+the `nat_degree` of its numerator and the `nat_degree` of its denominator. In particular,
+`int_degree 0 = 0`. -/
+def int_degree (x : ratfunc K) : ℤ := nat_degree x.num - nat_degree x.denom
```

Comment:
Can you add this to the main definition in the doc at the beginning of the file?

Reply:
Done.